bears/xml/XMLBear.py

```diff
@@ -1,5 +1,26 @@
+import os
+import requests
+
 from coalib.bearlib.abstractions.Lint import Lint
 from coalib.bears.LocalBear import LocalBear
+from coalib.misc.Shell import escape_path_argument
+from coalib.settings.Setting import path
+
+
+def path_or_url(type1, type2):
+    '''
+    Checks if the argument is a ``url`` or a ``path``.
+
+    :param type1:       path
+    :param type2:       str
+    :return:            path or str
+    '''
+    def _path_or_url(dtdvalid):
+        if dtdvalid is not None and os.path.exists(dtdvalid):
+            return type1
```

Comment:
Hm. I thought print would work.
You could try raise AssertionError(stuff to print)
That would force pytest to end as you're reading the error but will give you the debug info you want

-----Original Message-----
From: "vivek425ster" <notifications@github.com>
Sent: ‎13-‎03-‎2016 11:06
To: "coala-analyzer/coala-bears" <coala-bears@noreply.github.com>
Cc: "AbdealiJK" <abdealikothari@gmail.com>
Subject: Re: [coala-bears] XMLBear: Add args (#232)

In bears/xml/XMLBear.py:
>  from coalib.bearlib.abstractions.Lint import Lint
>  from coalib.bears.LocalBear import LocalBear
> +from coalib.misc.Shell import escape_path_argument
> +from coalib.settings.Setting import path
> +
> +
> +def path_or_url(type1, type2):
> +    '''
> +    Checks if the argument is a ``url`` or a ``path``.
> +
> +    :param type1:       path
> +    :param type2:       str
> +    :return:            path or str
> +    '''
> +    def _path_or_url(dtdvalid):
> +        if dtdvalid is not None and os.path.exists(dtdvalid):
> +            return type1
@AbdealiJK How to print it. A simple print statement isn't displaying anything in the error log of py.test
—
Reply to this email directly or view it on GitHub.